controllers/argocd/deployment.go

```diff
@@ -266,6 +266,10 @@ func getArgoServerCommand(cr *argoprojv1a1.ArgoCD) []string {
 	cmd = append(cmd, "--logformat")
 	cmd = append(cmd, getLogFormat(cr.Spec.Server.LogFormat))
 
+	if len(cr.Spec.Server.CommandArgs) > 0 {
+		cmd = append(cmd, cr.Spec.Server.CommandArgs...)
+	}
```

Comment:
I wonder whether we should check if one of the extra arguments shadows an argument that has been set by the Operator previously, and possibly log a warning or info message in that case?

I'm not sure how command line parsing would behave if there is the same option specified twice, but with a different value. Which one would take precedence? The first (set by the operator) or the second (set by the user)?

We should at least find out and document that behavior, so that there'll be no nasty surprises.

Reply:
After giving it a thought, I feel the better approach would be to error out when one of the extra arguments shadows an argument that has been set by the Operator previously. This will avoid unexpected consequences.